intera_interface/src/motion_interface/interaction_options.py

```diff
@@ -0,0 +1,244 @@
+                 K_impedance = None,
+                 max_impedance = None,
+                 K_nullspace = None,
+                 force_command= None,
+                 interaction_frame = None,
+                 endpoint_name = None,
+                 in_endpoint_frame = None,
+                 interaction_control_mode = None):
+        """
+        Create a interaction options object. All parameters are
+        optional. If ommitted or set to None, then use default value.
+        """
+        self._data = InteractionControlCommand()
+        self.set_header(header)
+        self.set_interaction_control_active(interaction_control_active)
+        self.set_K_impedance(K_impedance)
+        self.set_max_impedance(max_impedance)
+        self.set_K_nullspace(K_nullspace)
+        self.set_force_command(force_command)
+        self.set_interaction_frame(interaction_frame)
+        self.set_endpoint_name(endpoint_name)
+        self.set_in_endpoint_frame(in_endpoint_frame)
+        self.set_interaction_control_mode(interaction_control_mode)
+
+    def set_header(self, header = None):
+        if header is None:
+            self._data.header.stamp = rospy.get_rostime()
+            self._data.header.seq = 1
+            self._data.header.frame_id = "base"
+        else:
+            self._data.header = header
+
+    def set_interaction_control_active(self, interaction_active = None):
+        """
+        @param interaction_control_active:
+        --> None:  set it to True by default
+        --> [bool]: copy the element.
+        """
+        if interaction_active is None:
+            self.set_interaction_control_active(True) # default value
+        else:
+            self._data.interaction_control_active = interaction_active
+
+    def set_K_impedance(self, K_impedance = None):
+        """
+        @param K_impedance (Cartesian stiffness):
+        --> None:  populate with vector of default values
+        --> [float]:  copy all elements. Checks length.
+        """
+        if K_impedance is None:
+            self._data.K_impedance = [1300.0, 1300.0, 1300.0, 30.0, 30.0, 30.0]
+        elif len(K_impedance) < self.n_dim_cart:
+            rospy.logerr('The number of elements must be 6!')
+        elif len(K_impedance) == self.n_dim_cart:
+            self._data.K_impedance = deepcopy(K_impedance)
+        else:
+            rospy.logerr('Invalid input to K_impedance!')
+
+    def set_max_impedance(self, max_impedance = None):
+        """
+        @param max_impedance (impedance modulation state):
+        --> None:  populate with vector of default values (True)
+        --> bool:  set each dimension by the input boolean
+        --> [bool]:  copy all elements. Checks length.
+        """
+        if max_impedance is None:
+            self._data.max_impedance =  [True, True, True, True, True, True]
+        elif isinstance(max_impedance[0], bool) and len(max_impedance) == 1:
+            self._data.max_impedance = []
```

Comment:
Please forgive me for the pedantic pythonista comments :) 
`self._data.max_impedance = [max_impedance[0]]*self.n_dim_cart`